extensions/sql-database-projects/src/models/project.ts

```diff
@@ -160,23 +162,32 @@ export class Project {
 	 */
 	public async addSystemDatabaseReference(name: SystemDatabase): Promise<void> {
 		let uri: Uri;
+		let ssdtUri: Uri;
 		let dbName: string;
 		if (name === SystemDatabase.master) {
 			uri = this.getSystemDacpacUri(constants.masterDacpac);
+			ssdtUri = this.getSystemDacpacSsdtUri(constants.masterDacpac);
 			dbName = constants.master;
 		} else {
 			uri = this.getSystemDacpacUri(constants.msdbDacpac);
+			ssdtUri = this.getSystemDacpacSsdtUri(constants.msdbDacpac);
 			dbName = constants.msdb;
 		}
 
-		this.addDatabaseReference(uri, DatabaseReferenceLocation.differentDatabaseSameServer, true, dbName);
+		let systemDatabaseReferenceProjectEntry = new SystemDatabaseReferenceProjectEntry(uri, ssdtUri, dbName);
+		await this.addToProjFile(systemDatabaseReferenceProjectEntry);
 	}
 
 	public getSystemDacpacUri(dacpac: string): Uri {
 		let version = this.getProjectTargetPlatform();
 		return Uri.parse(path.join('$(NETCoreTargetsPath)', 'SystemDacpacs', version, dacpac));
 	}
 
+	public getSystemDacpacSsdtUri(dacpac: string): Uri {
+		let version = this.getProjectTargetPlatform();
+		return Uri.parse(path.join('$(DacPacRootPath)', 'Extensions', 'Microsoft', 'SQLDB', 'Extensions', 'SqlServer', version, 'SqlSchemas', dacpac));
+	}
+
 	public getProjectTargetPlatform(): string {
```

Comment:
the logic looks right but for precaution we should have test for this. (ignore if already added)